docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,353 @@
+| ------- | --------------- |
+| portability-policy | see [portability policy](#portability-policy) |
+| checked | [checked](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/checked-compiler-option) |
+| default-encoding | see [file encoding](#file-encoding) |
+| compilerversion | full version with SHA |
+| fallback-encoding | see see [file encoding](#file-encoding) |
+| define | [define](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/define-compiler-option) |
+| langversion | [langversion](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/langversion-compiler-option) |
+| nullable | [nullable](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references) |
+| optimize | [optimize](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/optimize-compiler-option) |
+| unsafe | [unsafe](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/unsafe-compiler-option) |
+
+#### Portability Policy
+
+Portability policy is derived from the [appconfig command option](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/appconfig-compiler-option). 
+
+Since appconfig is a pointer to a file not embedded in the PDB or PE, information that will directly impact the compilation is extracted. This is stored in a flag called `portability-policy` with a numeric value from \[0-3\]. This value directly correlates to reconstructing the [AssemblyPortabilityPolicy](https://github.com/dotnet/roslyn/blob/bdb3ece74c85892709f5e42ae7d67248999ecc3b/src/Compilers/Core/Portable/Desktop/AssemblyPortabilityPolicy.cs).
+    - 0 -> SuppressSilverlightPlatformAssembliesPortability = false, SuppressSilverlightLibraryAssembliesPortability = false
+    - 1 -> SuppressSilverlightPlatformAssembliesPortability = true, SuppressSilverlightLibraryAssembliesPortability = false
+    - 2 -> SuppressSilverlightPlatformAssembliesPortability = false, SuppressSilverlightLibraryAssembliesPortability = true
+    - 3 -> SuppressSilverlightPlatformAssembliesPortability = true, SuppressSilverlightLibraryAssembliesPortability = true
+
+#### File Encoding 
+
+Encoding will be stored in two keys: "default-encoding" and "fallback-encoding". 
+
+If "default-encoding" is present, it represents the forced encoding used, such as by passing in "codepage" to the command line. All files should be decoded using this codepage value.
+
+If "fallback-encoding" is present, it means that no default encoding was specified but an encoding was detected and used. The compiler [has logic](https://github.com/dotnet/roslyn/blob/462eac607741023e5c2d518ac1045f4c6dabd501/src/Compilers/Core/Portable/EncodedStringText.cs#L32) to determine an encoding is none is specified, so the value that was computed is stored so it can be reused in future compilations.
+
+Both values are codepage values for an encoding, such as `65001` for UTF-8. 
```

Comment:
>`65001` for UTF-8.  [](start = 57, length = 19)

I think it would be better to use [`Encoding.WebName`](https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.webname?view=netcore-3.1) because that's more standardized than code pages (code pages are Microsoft specific). Also document it's case-insensitive name.
See also [http://www.iana.org/assignments/character-sets/character-sets.xhtml](http://www.iana.org/assignments/character-sets/character-sets.xhtml)

Reply:
Done, I didn't know CodePage was Microsoft specific